### ISSUE TITLE
New Validate Option Flag to allow more granular control for entry amount check

### ISSUE THREAD

**Issue by hshih-lead** [wadearnold/ach#1450]
### ACH Version

All

### What were you trying to do?

**TLDR**
We would like to have the capability to allow $0 in entry amount while keeping the ValidAmountForCodes check.

**Current State**
The [ValidAmountForCodes](https://github.com/moov-io/ach/blob/fc02648af9169e9f4e172756234f7fd8099a8588/batch.go#L1037) check is performed on most of the batch headers (depending on the SEC code). It allows users to turn off the whole check via `validateOpts.AllowInvalidAmounts` flag.

**What We'd Like To See**
We're hoping a new option/flag could be added to only turn off the part where it [asserts the entry amount must NOT be $0](https://github.com/moov-io/ach/blob/fc02648af9169e9f4e172756234f7fd8099a8588/batch.go#L1066)
```
} else {
	if entry.Amount == 0 {
		return fieldError("Amount", ErrBatchAmountZero, entry.Amount)
	}
}
``` 

**Why We Need It**
We have seen in the Fed files we received containing entries with $0 that aren't prenotes or correction, but regular debit. In response, our users intend to send $0 returns. Though returning $0 is unusual, it's not strictly prohibited, since the processor is already accepting them today. We would like to support that use, and allow $0 to be the value but still keep the other checks in place (prenotes, Addenda98...etc).

### What did you expect to see?

An optional flag that functions similarly to other validation options, specifically to control the $0 check mentioned above.

Right now `ValidAmountForCodes` actually includes a few different checks. Having a more fine-grained control will give us more flexibility. Also a option flag would mean it's backward compatible and non-breaking, thus a low risk change.

**Comment by hshih-lead** [comment]
No tests added yet and I'm not sure I'm supposed to update just the spec and generate code, but something like the PR https://github.com/moov-io/ach/pull/1451 is what I imagine the implementation would be?
